WordPress/gutenberg
# Apply custom scroll style to fixed header block toolbar

## What?
This PR applies the custom scrolling style to the fixed block toolbar. I think this improves the appearance, especially on **Windows**.

> [!NOTE]
> This PR was originally to simply introduce a custom scrollbar. However, after some discussion, we have made a CSS adjustment in all editor instances to prevent the toolbar from shifting when a scrollbar appears or when a button is focused.

### Before (Chrome)

![chrome-before](https://github.com/WordPress/gutenberg/assets/54422211/398fdb68-1d4c-48c7-931a-82a89edfccd6)

### After (Chrome)

![chrome-after](https://github.com/WordPress/gutenberg/assets/54422211/1b0efc58-2563-4a1a-9649-5f619dc3f208)

## Why?

In the Chrome browser on Windows, the scrollbar has a _physical_ width/height. So the top toolbar buttons are pushed far up and look unnatural to me.

Additionally, this toolbar actually contains two `overflow` properties:

https://github.com/WordPress/gutenberg/blob/83e1a2c064c2233e8cd82dcccf4cd555c3497be1/packages/edit-site/src/components/header-edit-mode/style.scss#L206-L208

https://github.com/WordPress/gutenberg/blob/83e1a2c064c2233e8cd82dcccf4cd555c3497be1/packages/block-editor/src/components/block-toolbar/style.scss#L76-L77

In the case of the Site Editor, the `overfow` property value of the upper container (`.selected-block-tools-wrapper `) is `scroll` instead of `hidden`, so when element in the container overflow, the two scroll bars overlap, making it look even more unnatural.

![double-scrollbar](https://github.com/WordPress/gutenberg/assets/54422211/131568b4-24dd-4cca-8739-03fd6073f713)

## How?
First, I changed the `overflow `property of the upper container (`.selected-block-tools-wrapper`) from `scroll` to `hidden`. This will eliminate double scrollbars.

Additionally, to make this thick scrollbar appear more natural, I applied a `custom-scrollbars-on-hover` mixin to the inner scroll container (`.block-editor-block-toolbar`). If an element overflows, a scrollbar will appear without the need for mouseover. Personally, I feel that this is easier to understand than the behavior of disappearing the scroll bar when the mouse is out.

## Testing Instructions

- Open the Site Editor.
- Insert some block and select it. For example, image blocks are easier to test because they have a long block toolbar.
- Change the browser width in various browsers.
- For Mac OS, change the OS settings so that the scroll bar is always displayed.
- When the block toolbar is not fixed, the scrollbar should never be displayed.

## Screenshots or screencast 

Note: This shows scrollbar behavior in WindowsOS and Chrome.

### Before

https://github.com/WordPress/gutenberg/assets/54422211/a457de09-5b55-4cbd-8852-dc6352fd3e1a

### After

https://github.com/WordPress/gutenberg/assets/54422211/117a410c-150a-4ed6-b65a-1fbc28a82291
